### DISCUSSION TITLE
How to get access of REPL content & output?

### DISCUSSION THREAD

**Discussion by Linnore** [jupyterlite/jupyterlite#1441] [Q&A] [answered]
Hi, I am new to jupyterlite and find the REPL application very useful. However, unlike the jupyter lab application, REPL's cell content and outputs cannot be accessed by `jupyterlab/notebook`, by which I thought REPL cells could be identified and accessed as an active cell.

My goal is to let the iframe that containing an REPL cell communicates with the host page with the input and output of the REPL cell. 

Does anyone have a solution for this use case? I appreciate any help or ideas!

**Comment by jtpio** [comment]
> My goal is to let the iframe that containing an REPL cell communicates with the host page with the input and output of the REPL cell.

Maybe this could be achieved by following an approach similar to the one described in https://jupyterlite.readthedocs.io/en/stable/howto/configure/advanced/iframe.html?

And implementing a small extension that act as the interface between the host page and the inputs / outputs.

**Reply by Linnore** [reply]
Thanks for the advice! 

I am trying to use the same design of mini-extension as the example you mentioned. However, the problem I encounter is to get the REPL cell content "within the iframe" since I fail to find the right js interface to access the content.  

The example you mentioned only need the access of the global theme type, so it is very clear to use the "IThemeManager".

If the iframe embeds an entire jupyterlab/notebook, interfaces like "INotebookTracker" and "NotebookPanel" can be used to access cells then "ICellModel" to get the content. Sadly not the case for a minimal REPL cell...

**Reply by jtpio** [reply, marked as answer]
> If the iframe embeds an entire jupyterlab/notebook, interfaces like "INotebookTracker" and "NotebookPanel" can be used to access cells then "ICellModel" to get the content. Sadly not the case for a minimal REPL cell...

There is `IConsoleTracker` which is used in the REPL application here:

https://github.com/jupyterlite/jupyterlite/blob/b77efc239521666872535aa9006c4348d8bc6971/packages/repl-extension/src/index.ts#L123-L171
